iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp

```diff
@@ -463,10 +405,15 @@ static LogicalResult distributeCyclicallyToProcessors(
         cast<scf::ParallelOp>(serializeDimensionsFrom(rewriter, pLoopOp, 3));
     numLoops = 3;
   }
-  SmallVector<Value, 2> id(numLoops), count(numLoops);
-  getGPUProcessorIdsAndCounts<GPUIdOp, GPUCountOp>(pLoopOp.getLoc(), rewriter,
-                                                   numLoops, id, count);
-  return distributeCyclicallyToProcessors(rewriter, pLoopOp, id, count);
+  SmallVector<linalg::ProcInfo, 2> procInfo =
+      getGPUProcessorIdsAndCounts<GPUIdOp, GPUCountOp>(
+          rewriter, pLoopOp.getLoc(), numLoops);
+  SmallVector<Value, 2> ids(procInfo.size()), counts(procInfo.size());
+  for (auto info : enumerate(procInfo)) {
+    ids[info.index()] = info.value().procId;
+    counts[info.index()] = info.value().nprocs;
+  }
+  return distributeCyclicallyToProcessors(rewriter, pLoopOp, ids, counts);
```

Comment:
How about passing `ArrayRef<linalg::ProcInfo>` to `distributeCyclicallyToProcessors` directly, so we dont need to convert it into two vectors here?

I think we intent to get such information from Linalg level, so it makes sense to me.